text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+
+For Notifications, most likely we would have to do a custom solution instead of relying on GitHub notifications. There are a couple strategies we could follow:
+
+- Create issue templates based on project areas. These templates could auto-label issues
+- Create a single issue template similar to [`DefinitelyTyped`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/.github/ISSUE_TEMPLATE.md) that includes notifying owners so they trigger a notification
+  - This issue template explains that not notifying owners means that the issue will be ignored
+  - We could use a bot (like stale bot) to auto-close issues without responses in this format
+  - We could create a bot that infers project owners from issue titles if they write out the package name
+  
+This strategy bleeds into issue templates in that we would most likely want a specific structure around issue titles and contents for things like bugs, feature requests, and support.
+
+Pull request templates would ideally follow our current format where they have a corresponding issue and provide a high-level change log and steps to review.
+
+# Drawbacks
+
+The main disadvantages of using the strategies outlined above is that it forces our model into a monorepo. While this in and of itself is not a negative thing, it does shift a portion of our development time towards investing in tooling to help make working in a monorepo more enjoyable.
+
+Aspects of this include a custom CLI to help with working on a single component, in addition to tooling or strategies for scaling out our core tools so that they are either fast on a large number of files, or only run on files that have chanegd.
+
+In addition, while moving work to one project eliminates any confusion to issue creation, we run into a problem where everyone will get additional notifications that may not be related to their core project. The immediate action after this would be to ignore conversations that one is not actively participating in, but this could lead to issues not getting any response.
+
+There are ways to mitigate this, including issue templates, but ultimately scaling issues and pull requests in a monorepo with multiple teams/collaboratos is still an unknown.
+
+Finally, there is a concern around making everything under the Carbon umbrella public. The main use-case being: what if a team wants to use Carbon to build a component but is unable to make this component open-source? One solution could be the mirror our project to an internal GitHub Enterprise repo, but this could cause fragmentation in the ecosystem.
+
+# Alternatives
+
+The main alternative would be for **only** Carbon core to have a monorepo design. Given a desire to want to ship per-component packages, it is hard to see how we could scale doing repo-per-compnoent efforts. Particularly with respect to consistency between projects, keeping things up-to-date, and the experience of working with interlinked components, aka compound components.
```

Comment:
Another alternative would be for each implementation to also be a mono-repo ... the primary benefits here would be cutting down on the notifications, less tooling, fewer total files, while keeping all the remainder of the implementation benifits.

The primary downside would be the same slow fragmentation of the ecosystem as each monorepo diverged.